### PR TITLE
#11813: Fixing V No. 1

### PR DESCRIPTION
**Pull request by oarrietadotcms**
**ContentTypeFactoryImp.java**: Sanitize condition and search before adding to DotConnect.
**SQLUtil.java**: Separate Condition and Parameter evil words just as the sanitize methods.
**SQLUtilTest.java**: Add tests to check new condition and parameter evil words and methods.

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java [hunk 1]
```diff
@@ -526,12 +507,12 @@ private boolean dbDelete(ContentType type) throws DotDataException {
...
     }
     DotConnect dc = new DotConnect();
-    dc.setSQL(String.format(this.contentTypeSql.SELECT_QUERY_CONDITION, searchCondition.condition,orderBy));
+    dc.setSQL( String.format( this.contentTypeSql.SELECT_QUERY_CONDITION, SQLUtil.sanitizeCondition( searchCondition.condition ), orderBy ) );
     dc.setMaxRows(limit);
     dc.setStartRow(offset);
-    dc.addParam(searchCondition.search);
-    dc.addParam(searchCondition.search);
-    dc.addParam(searchCondition.search);
+    dc.addParam( SQLUtil.sanitizeParameter( searchCondition.search ) );
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: We should not need to sanitize a parameter in a prepared statement, the drive should take care of this for us, right?

Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: this is correct, not needed, only the sort by and searchCondition.condition replaced in the sql

Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: Good catch, if the parameter is with ?, the driver will escape invalid characters

Comment (oarrietadotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: you are right, as long as we use addParam method we don't need to sanitize it. Let me update the source code.

File: dotCMS/src/test/java/com/dotmarketing/common/util/SQLUtilTest.java [hunk 2]
```diff
@@ -550,5 +551,44 @@ public void testValidOr2SanitizeParameter() throws Exception {
         assertEquals("xxx nor xxx", s);
     }
 
+    @Test()
+    public void testValidCondition() throws Exception {
+
+        final String query = "structuretype = 1";
+        final String s = SQLUtil.sanitizeParameter( query );
```
Comment (dsilvam) [inline dotCMS/src/test/java/com/dotmarketing/common/util/SQLUtilTest.java]: should this call SQLUtil. sanitizeCondition instead?

File: dotCMS/src/test/java/com/dotmarketing/common/util/SQLUtilTest.java [hunk 3]
```diff
@@ -550,5 +551,44 @@ public void testValidOr2SanitizeParameter() throws Exception {
...
+
+        assertNotNull(s);
+        assertEquals(query, s);
+    }
+
+    @Test()
+    public void testInvalidCondition() throws Exception {
+
+        final String query = "and if(length(user())>0,sleep(10),2)";
+        final String s = SQLUtil.sanitizeParameter( query );
```
Comment (dsilvam) [inline dotCMS/src/test/java/com/dotmarketing/common/util/SQLUtilTest.java]: SQL.sanitizeCondition also?

**Comment by oarrietadotcms** [comment]
PR updated with code review feedback.

**Review by dsilvam — APPROVED** [review]

**Pull request merged**